Data points text labels in napari

sfmig opened the pull request:
Rebase after #434 

## Description

**What is this PR**

- [ ] Bug fix
- [ ] Addition of a new feature
- [ ] Other

**Why is this PR needed?**

**What does this PR do?**

## References

Please reference any existing issues/PRs that relate to this PR.

## How has this PR been tested?

Please explain how any new code has been tested, and how you have ensured that no existing functionality has changed.

## Is this a breaking change?

If this PR breaks any existing functionality, please explain how and why.

## Does this PR require an update to the documentation?

If any features have changed, or have been added. Please explain how the
documentation has been updated.

## Checklist:

- [ ] The code has been tested locally
- [ ] Tests have been added to cover all new functionality
- [ ] The documentation has been updated to reflect any changes
- [ ] The code has been formatted with [pre-commit](https://pre-commit.com/)

niksirbi reviewed (changes requested):
Hey @sfmig, I take no issue with your implementation, this is quite neat!

There is a catch however. Because the text is now exactly the same colour as the points (very aesthetically pleasing!), in many scenarios it's hard or impossible to read (because there is no contrast). This becomes more of a problem if you zoom in (some screenshots attached).

I think we must find a way around this, we shouldn't sacrifice readability.
Is there a way to perhaps align the text in another way (maybe shift its anchor point to the right of each point, or sth?

![napari_screenshot](https://github.com/user-attachments/assets/73641442-1a96-4695-aeb1-96c77c6dc353)
![napari_screenshot2](https://github.com/user-attachments/assets/fb0c4607-8b1c-49ab-9453-97af9db97826)

sfmig commented:
Yes, there is a way to add an offset and select a different anchor point for the text - I tried a bit but the labels were sometimes too far off from the points when zoomed in (I think because you set an absolute value of pixels) so I paused it. Will revisit and look in more detail - thanks for the attention to detail! 🙌✨

sfmig commented:
Thanks for the feedback @niksirbi !

The changes:
- we now set an anchor corner for the text (lower left) and a predefined offset (5 pixels). 
-  I added a clarification in the GUI because in some situations it may be confusing to identify which marker the text refers (see example below). But tbh this is not super vital since we plan to support filtering by keypoint and individual.
- I factored out the text properties that are "fixed" to reduce clutter and set them as defaults in the `PointsStyle` class, and added a `set_text_by` function to set the text following a property.
- I added a test for the `set_text_by` method.

![image](https://github.com/user-attachments/assets/f6186762-d69e-47b9-9a5f-9999aff6279f).

niksirbi commented on movement/napari/loader_widgets.py:
```diff
@@ -164,26 +165,29 @@ def _add_points_layer(self):
-        # Color the points by individual if there are multiple individuals
-        # Otherwise, color by keypoint
+        # Set markers' text
+        text_prop = "keypoint" if "keypoint" in self.props else "individual"
```
perhaps the text_prop should also be "individual" if "keypoint" is in props but there is only 1 unique value (i.e. centroid tracking).

Viewing the AEON data gave me this idea. If you only have 1 keypoint/individual, it's not very useful to have "centroid" written all over the place, it would be more useful to have the individual names.

Don't feel particularly strong about this, do as you wish.

sfmig replied:
Agreed, that was one of the aims of the PR basically as right now for multi-animal datasets, the display text also shows for all keypoints the individual name.

But I had missed this case, thanks for pointing it out. Will implement it and add a test!

niksirbi reviewed (approved):
Thanks @sfmig, it works now!

Approved, with two remaining comments (one of them optional).

sfmig commented on docs/source/user_guide/gui.md:
```diff
@@ -150,9 +150,9 @@ You will see a view similar to the one below:
-next to the points by enabling the `display text` option from the
-layer controls panel.
+the points are colour-coded by individual ID. If we enable the `display_text` option in the
+layer controls panel, the keypoint name will be displayed on the lower left corner of each point. If the
```
sorry dunno where my head was!

niksirbi replied:
that's way we have PR review 😉
